### DISCUSSION TITLE
Manually recreate request context?

### DISCUSSION THREAD

**Discussion by TitaniumHocker** [pallets/flask#5234] [Q&A]
Hello!

In my application I'm trying to implement something like "batch" requests to my API. In this case I wan't each request from the batch to be processed like separate request to API. To implement this I need to recreate request and application contexts.

I stuck on part where I need to pass an environ to `current_app.request_context` method.
Any suggestions?

**Comment by rahcusa** [comment]
@TitaniumHocker  I have a similar usecase, did you find a way to achieve this?

**Reply by TitaniumHocker** [reply]
nope

**Reply by rahcusa** [reply]
there is a repo here that tries to achieve this using multi-part requests
https://github.com/dtkav/flask-batch/blob/master/flask_batch/flask_batch.py
and another repo here
https://github.com/ramnes/flask-aggregator

**Reply by rahcusa** [reply]
I was able to get it working

```
from flask import current_app
from werkzeug.test import EnvironBuilder

def run_batch(): # view func
  builder = EnvironBuilder(environ_base=request.environ, method=..., json=..., path=...)
  new_environ = builder.get_environ()
  with current_app.request_context(new_environ):
    # rest of the logic is same as the one in flask_batch repo
```
